modules/sdk-core/src/bitgo/tss/common.ts

```diff
@@ -68,3 +70,43 @@ export async function sendSignatureShare(
     })
     .result();
 }
+
+/**
+ * Verifies that a TSS wallet signature was produced with the expected key and that the signed data contains the
+ * expected common keychain as well as the expected user and backup key ids
+ */
+export async function commonVerifyWalletSignature(params: {
+  walletSignature: openpgp.Key;
+  bitgoPub: openpgp.Key;
+  commonKeychain: string;
+  userKeyId: string;
+  backupKeyId: string;
+}): Promise<{ value: ArrayBuffer }[]> {
+  const { walletSignature, bitgoPub, commonKeychain, userKeyId, backupKeyId } = params;
+
+  // it appears that the result for the key that was used to sign is always at index 1
+  // at index 0 is the result for the key that was certified
+  const isValid = (await walletSignature.verifyPrimaryUser([bitgoPub]))[1].valid;
```

Comment:
To me, this seems a little risky, is there a more theoretically robust way to check exactly the certification coming from bitgo. I was thinking if we can loop on the pairs (keyid, valid) and find the pair where keyid matches bitgo's keyid, then check if valid is true for that pair.

Reply:
`.verifyPrimaryUser` is a openpgpjs function, with index 0 being the self-cert of the CA key, and index 1 being the cert of the sub-key (our proof in this context).

unless openpgp.js is bumped and they change this logic, then this should always be the case.    notice that `openpgp.Key` is passed in, so it will check against that public key.   

Are you making the argument that index 0 should be checked for validity?

Reply:
I wrote this code so that the signature has to be done with the bitgo pub to be considered valid. If I also consider index 0, I could pass in the user/ backup pub instead of the bitgo pub and would still consider the signature valid.

Reply:
index 0 should only show that the self-cert of the CA key is valid (gpg always includes a self-certification of the top CA key).

So by passing in bitgoPub, both indexes should return as valid.